src/extension/ScriptRunner.ts

```diff
@@ -173,4 +186,127 @@ export class ScriptRunner {
+
+                const descriptiveMessage = (topic.name + ':' + curMessage)
+                  .replace(pathRegex, '')
+                  .replace(locationRegex, '')
+                  .replace(/:{2,}/g, ':')
+
+                const position: Position = this.getPosition(
+                  location,
+                  path,
+                  confFile,
+                )
+                const range: Range = new Range(position, position)
+                const diagnostic = new Diagnostic(range, descriptiveMessage)
+
+                const diagnostics: Diagnostic[] = []
+                diagnostics.push(diagnostic)
+
+                const uri = Uri.parse(problemPath)
+
+                const curDiagnosticCollection: vscode.DiagnosticCollection =
+                  vscode.languages.createDiagnosticCollection()
+
+                curDiagnosticCollection.set(uri, diagnostics)
+                this.diagnosticCollection.push(curDiagnosticCollection)
+              })
+            }
+          })
+        })
+      })
+  }
+
+  private getResourceError(str: string): ResourceError {
+    const jsonContent: ResourceError = JSON.parse(str)
+
+    const resource_error: ResourceError = {
+      topics: [],
+    }
+
+    jsonContent.topics.forEach(topic => {
+      const curTopic: Topic = {
+        name: topic.name,
+        messages: [],
+      }
+      const curMessages: Message[] = []
+      topic.messages.forEach(message => {
+        const curMessage: Message = {
+          message: message.message,
+          location: message.location,
+        }
+        curMessages.push(curMessage)
+      })
+      curTopic.messages = curMessages
+      resource_error.topics.push(curTopic)
+    })
+
+    return resource_error
+  }
+
+  private getPosition(
+    location: RegExpExecArray | null,
+    path: RegExpExecArray | null,
+    confFilePath: string,
+  ): Position {
+    // TODO: if the path is to the configuration file, the position should be 0,0 anyway
+
+    if (path && !path.includes(confFilePath) && location) {
+      const rowPosition = parseInt(location[0].split(':')[0]) - 1
+      const colPosition = parseInt(location[0].split(':')[1]) - 1
+      return new Position(rowPosition, colPosition)
+    }
+    return new Position(0, 0)
+  }
+
+  private getPath(path: RegExpExecArray | null, confFile: string): string {
+    let workspaceFolderPath = ''
+    const workspaceFolders = workspace.workspaceFolders
+
+    if (workspaceFolders) {
+      workspaceFolderPath = workspaceFolders[0].uri.path + '/'
+    }
+
+    let pathToReturn = workspaceFolderPath + confFile
+
+    if (path) {
+      const relativePathRegex = /[0-9a-z_-]+\.[0-9a-z]+$/i
+      const relativePath = relativePathRegex.exec(path[0])
+
+      if (relativePath) {
+        const pathToCheck = (workspaceFolderPath + relativePath[0]).replace(
+          /\/{2,}"/i,
+          '/',
+        )
+
+        if (fs.existsSync(pathToCheck)) {
+          pathToReturn = pathToCheck
+        }
```

Comment:
1. I started to deal with relative paths in the resourse_error.js message, I'll try later to do it better.
2. I used a different API because the vscode one did not have a method to check if a file existed. I changed this to the vscode way, which is trying to read the file and catching an error if it cannot